components/renku_data_services/project/api.spec.yaml

```diff
@@ -0,0 +1,466 @@
+        description:
+          $ref: "#/components/schemas/Description"
+      required:
+        - "name"
+    ProjectPatch:
+      type: object
+      description: Patch of a project
+      additionalProperties: false
+      properties:
+        name:
+          $ref: "#/components/schemas/Name"
+        slug:
+          $ref: "#/components/schemas/Slug"
+        repositories:
+          $ref: "#/components/schemas/RepositoriesList"
+        visibility:
+          $ref: "#/components/schemas/Visibility"
+        description:
+          $ref: "#/components/schemas/Description"
+    ProjectPut:
+      description: A project to replace another one
+      type: object
+      additionalProperties: false
+      properties:
+        name:
+          $ref: "#/components/schemas/Name"
+        slug:
+          $ref: "#/components/schemas/Slug"
+        repositories:
+          $ref: "#/components/schemas/RepositoriesList"
+        visibility:
+          $ref: "#/components/schemas/Visibility"
+        description:
+          $ref: "#/components/schemas/Description"
+      required:
+        - "name"
+    Ulid:
+      description: ULID identifier
+      type: string
+      minLength: 26
+      maxLength: 26
+      pattern: "^[A-Z0-9]{26}$"  # This is case-insensitive
+      format: ulid
+    Name:
+      description: Renku project name
+      type: string
+      minLength: 1
+      maxLength: 99
+      example: "My Renku Project :)"
+    Slug:
+      description: A command-line friendly name for a project
+      type: string
+      minLength: 1
+      maxLength: 99
+      pattern: "^[a-z0-9]+[a-z0-9._-]*$"  # Cannot contain consecutive special characters or end in .git and .atom
+      example: "my-renku-project"
+    CreationDate:
+      description: The date and time the project was created
+      type: string
+      format: date-time
+      example: "2023-11-01T17:32:28Z"  # Time is always in UTC timezone
+    Description:
+      description: A description for project
+      type: string
+      maxLength: 5000
+    User:
+      description: A KeyCloak user
+      type: object
+      additionalProperties: false
+      properties:
+        id:
+          $ref: "#/components/schemas/UserId"
+      required:
+        - "id"
+      example:
+        id: "some-keycloak-user-id"
+    UserId:
+      description: KeyCloak user ID
+      type: string
+      pattern: "^[A-Za-z0-9-]+$"
+      minLength: 1
+      example: "123-keycloak-user-id-456"
+    RepositoriesList:
+      description: A list of repository URLs
+      type: array
+      items:
+        $ref: "#/components/schemas/GitUrl"
+      minItems: 0
+      uniqueItems: true
+      example:
+        - https://github.com/SwissDataScienceCenter/project-1.git
+        - git@github.com:SwissDataScienceCenter/project-2.git
+    GitUrl:
+      description: URL of a Git repository
+      type: string
+      format: uri  # TODO: This should support other formats like ``git@`` and ``https+git``
```

Comment:
As discussed in the meeting today, we **might** need a field in addition to the URI, which is backend type, which would be an enum of the backends we support (just "gitlab" at first, but later perhaps "github", "bitbucket", etc).

Reply:
Discussed in today's meeting, and we decided that we are not sure that we need this, so let us not make this change now. If it later turns out to be necessary, it should be easy to make the change at that time.